### DISCUSSION TITLE
Can we natively make a request to an external API ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [pallets/flask#5065] [Q&A] [answered]
Hi everyone,

Is it natively possible with flask to query an external api ? ( with a HTTP GET )

or do we need to use either another package or 
```python
from urllib.request import urlopen 
```
Thanks.

**Comment by RonnyPfannschmidt** [comment]
Flask is concerned with the server side, if you need a client as well those are provided by other packages

**Reply by BobSquarePants** [reply]
Indeed Flask is working backend (server side.) but server side it's also possible to make request to any external URI/URL.. The question was, is it natively possible with Flask ?

**Reply by ThiefMaster** [reply, marked as answer]
This is completely out of the scope of Flask. Use `requests` or a similar library to make outgoing requests. There is pretty much nothing in Flask (or werkzeug) that would make it logical to include an http client there...

**Reply by BobSquarePants** [reply]
Thanks @ThiefMaster 

It didn't came to my mind that a common http request was indeed part of an http client.
so indeed I'm looking over [requests](https://requests.readthedocs.io/en/latest/)

**Reply by davidism** [reply]
@BobSquarePants your last few questions here have not been particularly high quality. I suggest you take much more time on your own to learn about Python and Flask and research solutions before asking here again.
